### ISSUE TITLE
"No matching version found" when new version of Vault recently released

### ISSUE THREAD

**Issue by bkonicek-calm** [eLco/setup-vault#4]
We've observed the following error when using this action whenever there has been a recent release of Vault within an hour or so of running the action. This appears to be caused because the action looks up the latest release from the Vault Github repo's releases, but attempts to download the release from https://releases.hashicorp.com/vault/. There seems to be a delay in Hashicorp's release lifecycle between the release being published on Github, and the matching packages being uploaded to `releases.hashicorp.com`.

Perhaps there is some way to fall back to trying one version prior to the latest if the `latest` release download fails.

```
Run eLco/setup-vault@v1.0.2
  with:
    vault_version: latest
    github_token: ***
Get release info for latest version
Error: Error: No matching version found
Error: No matching version found
```

**Comment by leiftaylorcb** [comment]
I am also seeing this issue:
```
Run eLco/setup-vault@v1
Get release info for latest version
(node:1839) [DEP000[5](https://github.com/Connected2FiberTeam/cb-iac/actions/runs/6804229593/job/18501292234#step:2:6)] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Error: Error: No matching version found
Error: No matching version found
```

```
      - uses: gliech/create-github-secret-action@v1
        with:
          location: Connected2FiberTeam
          name: VAULT_TOKEN
          value: ${{ env.NEW_TOKEN }}
          org_visibility: private
          pa_token: ${{ secrets.GIT_ADMIN_PAT }}
```

**Comment by bkonicek-calm** [comment]
Yeah the latest github release is `1.15.2` 
![image](https://github.com/eLco/setup-vault/assets/127137859/edcd84a6-392b-4245-9d68-22d13637ff51)

but the latest available on releases site is `1.15.1` 
![image](https://github.com/eLco/setup-vault/assets/127137859/96e6faa4-0595-471f-a9aa-4ce3b2a9449f)

We've pinned our action version at `1.15.1` for now and it's working but ideally would like to always install latest

**Comment by eLco** [comment]
> I am also seeing this issue:
> 
> ```
> Run eLco/setup-vault@v1
> Get release info for latest version
> (node:1839) [DEP000[5](https://github.com/Connected2FiberTeam/cb-iac/actions/runs/6804229593/job/18501292234#step:2:6)] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
> (Use `node --trace-deprecation ...` to show where the warning was created)
> Error: Error: No matching version found
> Error: No matching version found
> ```
> 
> ```
>       - uses: gliech/create-github-secret-action@v1
>         with:
>           location: Connected2FiberTeam
>           name: VAULT_TOKEN
>           value: ${{ env.NEW_TOKEN }}
>           org_visibility: private
>           pa_token: ${{ secrets.GIT_ADMIN_PAT }}
> ```

Hi @leiftaylorcb I’m sorry I didn’t set v1 tag to the latest version containing fix for Buffer deprecation. It should be fixed now.
